src/main/java/edu/ie3/datamodel/utils/validation/ConnectorValidationUtils.java

```diff
@@ -38,21 +50,41 @@ private ConnectorValidationUtils() {
    * the checking task, based on the class of the given object.
    *
    * @param connector Connector to validate
-   * @throws edu.ie3.datamodel.exceptions.NotImplementedException if an unknown class is handed in
+   * @return a list of try objects either containing a {@link InvalidEntityException} or an empty
+   *     Success
    */
-  protected static void check(ConnectorInput connector) {
-    checkNonNull(connector, "a connector");
-    connectsDifferentNodes(connector);
+  protected static List<Try<Void, InvalidEntityException>> check(ConnectorInput connector) {
+    try {
+      checkNonNull(connector, "a connector");
+    } catch (InvalidEntityException e) {
+      return List.of(
+          new Failure<>(
+              new InvalidEntityException(
+                  "Validation not possible because received object {" + connector + "} was null",
+                  e)));
```

Comment:
This exception is built a lot of times, and SQ is already complaining about the duplicate String literals. Maybe creating a small static method for this would make sense, maybe in `ValidationUtils`?